### DISCUSSION TITLE
Questions about resource loading in the chat interface

### DISCUSSION THREAD

**Discussion by muhanstudio** [open-webui/open-webui#7691] [General]
A long time ago, I encountered an issue where images in chat caused switching and browsing to become exceptionally laggy. I reported this issue, but it was converted into a discussion, and no one has replied for a long time.

[Link to the discussion](https://github.com/open-webui/open-webui/discussions/5375#discussion-7172826)

In my opinion, this may not be an easy issue to overlook. In any similar projects, when switching, the interface DOM and text are loaded first, while images are left as placeholders and are loaded only when ready. However, in OpenWebUI, when switching conversations, the program must download all image resources from the conversation before it starts switching. Until this action is completed, the program enters a state of apparent freeze. I believe this must affect many people, especially those deploying on public servers with limited bandwidth.

I don't understand why no one has responded or discussed this. With today's imaging technology, a single photo is usually over 4MB. Having just one image in a conversation can cause significant lag, not to mention asking the AI for three or four images in a row, which results in extreme lag. This should be a matter of concern. Images in the chat interface should be linked rather than treated as essential resources for the rendering thread.

All I really need is for someone to reply, letting me know that I am not the only one experiencing this issue.

Thank you again to the hardworking authors for bringing this amazing open-source project to life.

**Comment by a15355447898a** [comment]
I have the same question.

**Comment by tjbck** [comment]
Will add a loading indicator, stay tuned.
